packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -310,20 +310,39 @@ export class TrimergeClient<
       ),
   };
 
-  getCommitDoc(ref: string): CommitDoc<SavedDoc, CommitMetadata> {
-    const doc = this.docCache.get(ref);
-    if (doc !== undefined) {
-      return doc;
+  getCommitDoc(headRef: string): CommitDoc<SavedDoc, CommitMetadata> {
+    let baseDoc: CommitDoc<SavedDoc, CommitMetadata> | undefined;
+    const commitWalk: string[] = [];
+    let currentCommitRef: string | undefined = headRef;
+    while (!baseDoc && currentCommitRef !== undefined) {
+      if (this.docCache.has(currentCommitRef)) {
+        baseDoc = this.docCache.get(currentCommitRef);
+        break;
+      }
+
+      commitWalk.push(currentCommitRef);
+      const commit = this.getCommit(currentCommitRef);
+      currentCommitRef = commit.baseRef;
+    }
+
+    let resultDoc = baseDoc;
```

Comment:
Maybe just recycle the same variable since there's no need to hold onto base doc